plugin.go

```diff
@@ -62,23 +65,34 @@ type (
 	dispatchFunc func(op PluginOpcode, index int32, value int64, ptr unsafe.Pointer, opt float32) int64
 )
 
-func (d Dispatcher) dispatchFunc(params []*Parameter) dispatchFunc {
+func (d Dispatcher) dispatchFunc(p Plugin) dispatchFunc {
 	return func(op PluginOpcode, index int32, value int64, ptr unsafe.Pointer, opt float32) int64 {
 		switch op {
 		case plugGetParamName:
-			s := (*ascii8)(ptr)
-			copyASCII(s[:], params[index].Name)
+			s := (*ascii32)(ptr)
+			copyASCII(s[:], p.Parameters[index].Name)
 		case plugGetParamDisplay:
 			s := (*ascii8)(ptr)
-			copyASCII(s[:], params[index].ValueLabel)
+			copyASCII(s[:], p.Parameters[index].ValueLabel)
 		case plugGetParamLabel:
 			s := (*ascii8)(ptr)
-			copyASCII(s[:], params[index].Unit)
+			copyASCII(s[:], p.Parameters[index].Unit)
 		case plugSetBufferSize:
 			if d.SetBufferSizeFunc == nil {
 				return 0
 			}
 			d.SetBufferSizeFunc(int(value))
+		case PlugGetPluginName:
+			s := (*ascii64)(ptr)
```

Comment:
According to SDK spec, this should also be 32. I just googled header files on github and found [this](https://github.com/R-Tur/VST_SDK_2.4/blob/4610873fa433d263a5c4a68b5b8b59ffcf6ca6db/pluginterfaces/vst2.x/aeffect.h#L320) repo.